### ISSUE TITLE
mason installation, binary maybe misplaced

### ISSUE THREAD

**Issue by lucaferranti** [chapel-lang/chapel#19910]
### Summary of Problem

When I installed chapel, the binary of `chpl` was placed in `/usr/local/bin` (as expected). When I installed mason, the binary was placed in `chapel-1.26.0/bin/linux64-x86_64`. 

What confused me is that the docs says [here](https://chapel-lang.org/docs/tools/mason/mason.html#installation-instructions)

> This installs mason in the same place as the chapel compiler (chpl) so that mason can be used anywhere in the user’s file system.

which was not the case in my situation.

I am no make wizard, but maybe the `Makefile` for `mason` (and other tools) could check if a `configured-prefix` has been generated and copy the binary to that path? Alternatively just adding a note in the docs would help too (happy to open a PR myself if it helps).

### Steps to Reproduce

I installed chapel compiler with 

```
cd chapel-1.26.0
./configure
make
make install
```
this placed the binary of the compiler into `usr/local/bin`.

I installed mason with

```
cd chapel-1.26.0
make mason
```

in case it's relevant I did **not** close the terminal session between chapel and mason installation. This placed the binary for mason under `chapel-1.26.0/bin/linux64-x86_64`.

### Configuration Information

- Output of `chpl --version`:
```
chpl version 1.26.0
  built with LLVM version 12.0.0
Copyright 2020-2022 Hewlett Packard Enterprise Development LP
Copyright 2004-2019 Cray Inc.
(See LICENSE file for more details)
```
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: llvm
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: system
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
```
gcc (Ubuntu 10.3.0-1ubuntu1) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
- (For Cray systems only) Output of `module list`:

**Comment by bradcray** [comment]
@lucaferranti : Thanks for reporting this.  From what I'm seeing, it looks like a simple bug on our part that our testing failed to catch for us.  The build of `mason` used to create `mason` in `tools/mason/`, which is where our install script tries to install it from.  But the build logic was updated at some point to create `mason` directly in `bin/...` and the install logic wasn't updated to reflect that.  I expect that the fix will be trivial and that this will be included in our summer release.

**Comment by bradcray** [comment]
@lucaferranti : Thanks again for filing this.  I marked this as resolved in #19911 and was coming over to post the preceding, but upon re-reading the OP am realizing that strictly speaking, I didn't completely resolve it, so am re-opening.  That said, I want to get a sense of what you think the right way forward is here.

**Comment by bradcray** [comment]
[Oops, wasn't expecting the "reopen" button to post that comment, which I was still writing.  Continuing...]

More specifically:  The bug I found and fixed was that `make install` wasn't installing a built `mason` into `/usr/local/bin/` (say) in the following sequence:

```bash
$ ./configure
$ make
$ make mason
$ make install
```

But re-reading, the OP, I'm realizing that I mis-internalized it and that your sequence was actually:

```
$ ./configure
$ make
$ make install
$ make mason
```

and that the documentation does suggest that this will put `mason` into `/usr/local/bin/` as well.  It seems we have two choices:

* My instinct is that things shouldn't be put into the installation directory without an explicit `make install` step, in which case the current (post- #19911 ) behavior is appropriate, such that the documentation should be updated (where I read the current wording as the author saying "It's too much trouble to describe `$CHPL_HOME/bin/whatever-goes-here/` accurately, so I'll use this roundabout phrasing instead.").  By analogy, the `make` of `chpl` itself doesn't put it into `/usr/local/bin`, though it arguably could since the `./configure` has already been run.

* The alternative would be to make an argument that making `mason` put it into the configured directory for some reason.  I'm not aware of a precedent for that in other Makefile/autoconf/autoconf-like setups, but may just be unaware.

Would you let me know whether you agree with / prefer my (first) interpretation, or have an argument for taking the second?

(And if you do prefer the first:  You offered to make a PR adjusting the docs, and I would appreciate that if you wanted to, but would be happy to take that on ourselves if you'd just as soon not bother.  I expect that the key will be to see how we describe the `$CHPL_HOME/bin/this-other-stuff/` directory elsewhere and lean on that description, ideally without creating a maintenance challenge if we keep evolving the naming of the subdirectory).

**Comment by lucaferranti** [comment]
hi @bradcray, sorry for the delay in answering this.

so I think after #19924 this issue, which originally regarded mason, can be closed. Regarding the extra points

1. I removed `chpldoc`  from my `/usr/local/bin` and just tried the following while being in `chapel-1.26.0`
```
./configure
make chpldoc
make install
```
this did not move the binary into `/usr/local/bin` . Moreover, I just tried to use chpldoc (well after manually copying the binary into `/usr/local/bin`) to document a library and when doing `mason doc` I got the following error
```
cannot stat '/usr/local/share/chapel/1.26/third-party/chpl-venv/chpldoc-sphinx-project/*
```
after some digging I fixed this by manually copying `chapel-1.26.0/third-party/chpl-venv` into `usr/local/share/chapel/1.26`. am I doing some trivial mistake or is this something worth opening an issue for further investigation?

2. Good point, I was basically asking what's the neatest way to get a full battery included installation of compiler and all tools. Your one line solution seems indeed pretty handy.

**Comment by bradcray** [comment]
> this did not move the binary into /usr/local/bin .

Oh gosh, you're right of course.  Checking what I was referring to above, it seems that I must have been quickly reading a "TODO" doc saying that we _should_ install `chpldoc` as though it said that it was installing it.  Duh.

So, yes, I think this is worth opening an issue for.  Checking, I found this one https://github.com/chapel-lang/chapel/issues/7574 which arguably subsumes it, but I think it'd be smart to open it as its own issue to disentangle it, particularly if it's something you'd like to champion.

Meanwhile, I will close this one, as I think we're in agreement that it's served its purpose.  Thanks!
